Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -46,25 +46,25 @@ void gdsp_ifx_init()
 		g_dsp.ifx_regs[i] = 0;
 	}
 
-	g_dsp.mbox[0] = 0;
-	g_dsp.mbox[1] = 0;
+	g_dsp.mbox[0].store(0);
+	g_dsp.mbox[1].store(0);
 }
 
 u32 gdsp_mbox_peek(u8 mbx)
 {
-	return Common::AtomicLoad(g_dsp.mbox[mbx]);
+	return g_dsp.mbox[mbx].load();
 }
 
 void gdsp_mbox_write_h(u8 mbx, u16 val)
 {
-	const u32 new_value = (Common::AtomicLoadAcquire(g_dsp.mbox[mbx]) & 0xffff) | (val << 16);
-	Common::AtomicStoreRelease(g_dsp.mbox[mbx], new_value & ~0x80000000);
+	const u32 new_value = (g_dsp.mbox[mbx].load() & 0xffff) | (val << 16);
```

Comment:
Are the 16-bit halves ever used together atomically? If not, just split them up. If yes, is there a better way to do it than this?

```c++
u32 desired, expected = mbox.load();
do
{
    desired = (val << 16) | (expected & 0xFFFF);
    expected = std::atomic_compare_exchange_weak(mbox, expected, desired);
} while (expected != desired);
```